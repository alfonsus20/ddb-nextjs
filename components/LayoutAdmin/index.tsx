import {
  Badge,
  Box,
  Button,
  ButtonGroup,
  Container,
  Flex,
  Table,
  TableCaption,
  TableContainer,
  Tbody,
  Td,
  Text,
  Tfoot,
  Th,
  Thead,
  Tr,
  VStack,
} from "@chakra-ui/react";
import Link from "next/link";
import React from "react";

type Props = {
  children: React.ReactNode;
};

const LayoutAdmin = ({ children }: Props) => {
  return (
    <Container maxW="container.2xl" pt={4} pb={12} px={12}>
      <Text fontSize="3xl" fontWeight="bold" mb={4}>
        Halaman Admin
      </Text>
      <Flex gap={8}>
        <Box flex="0 0 300px">
          <VStack spacing={4}>
            <Link href="/admin/mahasiswa" passHref>
              <Button width="full">Kelola Mahasiswa</Button>
            </Link>
            <Link href="/admin/berita" passHref>
              <Button width="full">Kelola Berita</Button>
            </Link>
          </VStack>
        </Box>
        <Box flex="1 1 auto">
          <TableContainer>
            <Table>
              <Thead>
                <Tr>
                  <Th>No.</Th>
                  <Th>Nama</Th>
                  <Th>Status</Th>
                  <Th>Terverifikasi</Th>
                  <Th>Action</Th>
                </Tr>
              </Thead>
              <Tbody>
                {[...Array(10)].map((_, idx) => (
                  <Tr key={idx}>
                    <Td>{idx + 1}.</Td>
                    <Td>Alfonsus Avianto Chandrawan</Td>
                    <Td>
                      <Badge variant="solid" colorScheme="cyan">
                        Mahasiswa Aktif
                      </Badge>
                    </Td>
                    <Td>
                      <Badge variant="solid" colorScheme="green">
                        Sudah
                      </Badge>
                    </Td>
                    <Td>
                      <ButtonGroup>
                        <Button colorScheme="blue">Edit</Button>
                        <Button colorScheme="red">Delete</Button>
                      </ButtonGroup>
                    </Td>
                  </Tr>
                ))}
              </Tbody>
            </Table>
          </TableContainer>
        </Box>
      </Flex>
    </Container>
  );
};

export default LayoutAdmin;
