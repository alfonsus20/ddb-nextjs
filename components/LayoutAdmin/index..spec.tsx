import { render } from "@testing-library/react";
import LayoutAdmin from ".";

describe("Layout Admin unit test", () => {
  it("should render correcly", () => {
    const container = render(
      <LayoutAdmin title="Testing">
        <p>masuk</p>
      </LayoutAdmin>
    );

    expect(container).toMatchSnapshot();
  });
});
