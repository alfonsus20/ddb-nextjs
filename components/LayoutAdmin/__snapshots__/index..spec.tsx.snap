// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Admin unit test should render correcly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="chakra-container css-1bwwudj"
      >
        <h2
          class="chakra-heading css-f4qdoo"
        >
          Testing
        </h2>
        <div
          class="css-1bvc4cc"
        />
         
        <button
          aria-label="open drawer"
          class="chakra-button css-15cfyfj"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-13otjrl"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            />
          </svg>
        </button>
        <div
          class="css-11qjisw"
        >
          <p>
            masuk
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-container css-1bwwudj"
    >
      <h2
        class="chakra-heading css-f4qdoo"
      >
        Testing
      </h2>
      <div
        class="css-1bvc4cc"
      />
       
      <button
        aria-label="open drawer"
        class="chakra-button css-15cfyfj"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-13otjrl"
          fill="currentColor"
          focusable="false"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          />
        </svg>
      </button>
      <div
        class="css-11qjisw"
      >
        <p>
          masuk
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
